eve/main.yml

```diff
@@ -39,3 +41,34 @@ stages:
             npm run flow &&
             npm run lint
           haltOnFailure: false
+  post-merge:
+    worker:
+      type: local
+    steps:
+      - TriggerStages:
+          name: Trigger storybook_deploy
+          stage_names:
+            - storybook_deploy
+          haltOnFailure: true
+  storybook_deploy:
+    worker:
+      type: kube_pod
+      path: eve/workers/pod-unit-tests/pod.yaml
+      images:
+        docker-unit-tests:
+          context: '.'
+          dockerfile: eve/workers/pod-unit-tests/Dockerfile
+    steps:
+      - Git: &git_pull
```

Comment:
you can replace this whole step as
`- Git: *git_pull` and that's it, remove all the other lines.

here you're creating two reference with the `&` character, which makes your `yaml` wrong